### ISSUE TITLE
[x264] build failure

### ISSUE THREAD

**Issue by fawdlstty** [microsoft/vcpkg#13603]
**Host Environment**
- OS: Win10 1909 (18363.900)
- Compiler: vs2019 community 16.7.2

**To Reproduce**
`vcpkg install x264:x86-windows-static x264:x64-windows-static`

**Failure logs** 
```
C:\Users\fawdlstty>vcpkg install x264:x86-windows-static x264:x64-windows-static
Computing installation plan...
The following packages will be built and installed:
    x264[core]:x64-windows-static
    x264[core]:x86-windows-static
Detecting compiler hash for triplet x64-windows-static...
Detecting compiler hash for triplet x86-windows-static...
Starting package 1/2: x264:x64-windows-static
Building package x264[core]:x64-windows-static...
Could not locate cached archive: C:\Users\fawdlstty\AppData\Local\vcpkg/archives\2a\2a00d5aacbc6712bbb7a77ba71f56a065e02e59a.zip
-- Using cached D:/gitee/vcpkg/downloads/mirror-x264-303c484ec828ed0d8bfe743500e70314d026c3bd.tar.gz
-- Cleaning sources at D:/gitee/vcpkg/buildtrees/x264/src/14d026c3bd-bca2ebf740.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/gitee/vcpkg/downloads/mirror-x264-303c484ec828ed0d8bfe743500e70314d026c3bd.tar.gz
-- Applying patch uwp-cflags.patch
-- Using source at D:/gitee/vcpkg/buildtrees/x264/src/14d026c3bd-bca2ebf740.clean
-- Using msys root at D:/gitee/vcpkg/downloads/tools/msys2/77f3fd5f29257d6b
-- Configuring x64-windows-static-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:85 (message):
    Command failed: D:/gitee/vcpkg/downloads/tools/msys2/77f3fd5f29257d6b/usr/bin/bash.exe --noprofile --norc --debug -c "V=1 CPP='cl.exe -nologo -E' CC='cl.exe -nologo' CXX='cl.exe -nologo' AR='lib.exe -verbose' LD='link.exe -verbose' RANLIB=':' CCAS=':' STRIP=':' NM='dumpbin.exe -symbols -headers' DLLTOOL='link.exe -verbose -dll' ./../src/14d026c3bd-bca2ebf740.clean/configure  --build=x86_64-pc-mingw32 --host=x86_64-pc-mingw32 --enable-strip --disable-lavf --disable-swscale --disable-avs --disable-ffms --disable-gpac --disable-lsmash --disable-asm --enable-debug --disable-silent-rules --verbose --disable-shared --enable-static --prefix=/D/gitee/vcpkg/installed/x64-windows-static/debug"
    Working Directory: D:/gitee/vcpkg/buildtrees/x264/x64-windows-static-dbg
    Error code: 1
    See logs for more information:
      D:\gitee\vcpkg\buildtrees\x264\config-x64-windows-static-dbg-out.log
      D:\gitee\vcpkg\buildtrees\x264\config-x64-windows-static-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:540 (vcpkg_execute_required_process)
  ports/x264/portfile.cmake:28 (vcpkg_configure_make)
  scripts/ports.cmake:79 (include)


Error: Building package x264:x64-windows-static failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: x264:x64-windows-static
  Vcpkg version: 2020.06.15-nohash

Additionally, attach any relevant sections from the log files above.

C:\Users\fawdlstty>
```

`D:\gitee\vcpkg\buildtrees\x264\config-x64-windows-static-dbg-out.log`:
```
Unknown option --build=x86_64-pc-mingw32, ignored
Unknown option --disable-silent-rules, ignored
Unknown option --verbose, ignored
Unknown option --disable-shared, ignored
Unknown system , edit the configure
```

`D:\gitee\vcpkg\buildtrees\x264\config-x64-windows-static-dbg-err.log`:
```
./../src/14d026c3bd-bca2ebf740.clean/configure: line 327: rm: command not found
./../src/14d026c3bd-bca2ebf740.clean/configure: line 328: rm: command not found
./../src/14d026c3bd-bca2ebf740.clean/configure: line 344: dirname: command not found
./../src/14d026c3bd-bca2ebf740.clean/configure: line 341: dirname: command not found
./../src/14d026c3bd-bca2ebf740.clean/configure: line 345: grep: command not found
./../src/14d026c3bd-bca2ebf740.clean/configure: line 554: /config.sub: No such file or directory
./../src/14d026c3bd-bca2ebf740.clean/configure: line 565: basename: command not found
./../src/14d026c3bd-bca2ebf740.clean/configure: line 148: rm: command not found
./../src/14d026c3bd-bca2ebf740.clean/configure: line 167: cat: command not found
./../src/14d026c3bd-bca2ebf740.clean/configure: line 171: cat: command not found
./../src/14d026c3bd-bca2ebf740.clean/configure: line 1: rm: command not found
```

**Additional context**
I try to remove `downloads\tools\msys2` directory and rebuild, still failed.

**Comment by bogdanlytvynovskyi** [comment]
Same issue easily reproducible. Looks like msys env is not complete

**Comment by bogdanlytvynovskyi** [comment]
A bit of digging shows that just launching bash does not add proper directory to path

**Comment by bogdanlytvynovskyi** [comment]
Apparently the culprit is this line in vcpkg_configure_make.cmake:

`        string(REPLACE ";$ENV{SystemRoot}\\System32;" ";${MSYS_ROOT}/usr/share/automake-1.16;${MSYS_ROOT}/usr/bin;$ENV{SystemRoot}\\System32;" NEWPATH "$ENV{PATH}")
 `

Cmake is case sensitive in replace, so if your env is like mine, then your path contains C:\Windows\system32 instead of C:\Windows\System32, and replace does nothing. This results in bash not having it's internal tools in path

**Comment by bogdanlytvynovskyi** [comment]
And while above mentioned helps, install step will still fail with some cryptic error messages

**Comment by bogdanlytvynovskyi** [comment]
Judging by changes to msys consumption summon for help: @Neumann-A

**Comment by Neumann-A** [comment]
a) this is a duplicate of another issue
b) there is already a PR fixing it.

**Comment by NancyLi1013** [comment]
Thanks for posting this issue.
It's duplicated to #13327. It will be fixed by PR #13450.
